spec/system/email_change_spec.rb

```diff
@@ -60,9 +62,9 @@ def get_link_from_email(type)
 
     find("button[type=submit]").click
 
-    try_until_success { expect(current_url).to match("/u/#{user.username}/preferences/account") }
+    expect(page).to have_current_path("/u/#{user.username}/preferences/account")
 
-    expect(user.reload.email).to eq(new_email)
+    try_until_success { expect(user.reload.email).to eq(new_email) }
```

Comment:
Shouldn't this one also use `expect(user_preferences_page).to have_primary_email(new_email)` ?